contracts/peripheral/YieldDirectorV2.sol

```diff
@@ -0,0 +1,441 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.10;
+
+import {IERC20} from "../interfaces/IERC20.sol";
+import {IsOHM} from "../interfaces/IsOHM.sol";
+import {IgOHM} from "../interfaces/IgOHM.sol";
+import {IStaking} from "../interfaces/IStaking.sol";
+import {SafeERC20} from "../libraries/SafeERC20.sol";
+import {YieldSplitter} from "../types/YieldSplitter.sol";
+import {OlympusAccessControlled, IOlympusAuthority} from "../types/OlympusAccessControlled.sol";
+
+/**
+    @title YieldDirectorV2 (codename Tyche) 
+    @notice This contract allows donors to deposit their gOHM and donate their rebases
+            to any address. Donors will be able to withdraw the sOHM equivalent of their principal
+            gOHM at any time. Donation recipients can also redeem accrued rebases at any time.
+ */
+contract YieldDirectorV2 is YieldSplitter, OlympusAccessControlled {
+    using SafeERC20 for IERC20;
+
+    address public immutable sOHM;
+    IStaking public immutable staking;
+    mapping(address => uint256[]) public recipientIds;
+    mapping(uint256 => address) internal recipientLookup;
+
+    bool public depositDisabled;
+    bool public withdrawDisabled;
+    bool public redeemDisabled;
+
+    event Deposited(address indexed donor_, address indexed recipient_, uint256 amount_);
+    event DepositUpdated(address indexed donor_, address indexed recipient_, uint256 amount_);
```

Comment:
Why not have both deposit and depositUpdate as same event? They are both essentially a user adding money into a deposit im not seeing the need for why they need seperate events. I feel like update implies some configuration or parameter change when its just adding more money in.

Reply:
done